crates/proof/proof-interop/src/provider.rs

```diff
@@ -136,9 +143,37 @@ where
             return Err(OracleProviderError::BlockNumberPastHead(number, header.number));
         }
 
+        // Set the chain ID for the trie hints, and explicitly drop the lock.
+        let mut chain_id_lock = self.chain_id.write();
+        *chain_id_lock = Some(chain_id);
+        drop(chain_id_lock);
```

Comment:
Using this spin-locked variable in the provider itself to communicate with the `TrieHinter` implementation is ugly, but serves to not break the `TrieHinter` interface for now. I'll throw up a ticket to improve this API.